File: CTF_Writeups/scripts_go/MetaCTF_Physics_Notes.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func extractFlag(text string) (string, error) {
	rawLines := strings.Split(text, "\n")
	lines := make([]string, 0, len(rawLines))
	for _, line := range rawLines {
		line = strings.TrimRight(line, "\r")
		if line == "" {
			continue
		}
		lines = append(lines, line)
	}

	if len(lines) < 15 {
		return "", fmt.Errorf("expected at least 15 non-empty lines")
	}

	var b strings.Builder
	for i := 0; i < 14; i++ {
		r := []rune(lines[i])
		b.WriteRune(r[0])
		b.WriteRune(r[len(r)-1])
	}

	last := []rune(lines[14])
	b.WriteRune(last[0])
	return b.String(), nil
}

func main() {
	data, err := os.ReadFile("CTF_Writeups/files/Physics Notes/notes.txt")
	if err != nil {
		fmt.Println("read error:", err)
		os.Exit(1)
	}

	flag, err := extractFlag(string(data))
	if err != nil {
		fmt.Println("extract error:", err)
		os.Exit(1)
	}

	fmt.Println(flag)
}

File: scripts_python/MetaCTF_Physics_Notes.py
from pathlib import Path


def extract_flag(text: str) -> str:
    lines = [line for line in text.splitlines() if line]
    if len(lines) < 15:
        raise ValueError("expected at least 15 non-empty lines")

    parts = [line[0] + line[-1] for line in lines[:14]]
    parts.append(lines[14][0])
    return "".join(parts)


def main() -> None:
    notes_path = Path("CTF_Writeups/files/Physics Notes/notes.txt")
    text = notes_path.read_text(encoding="utf-8")
    flag = extract_flag(text)
    print(flag)


if __name__ == "__main__":
    main()
